src/test/java/com/squareup/kotlinpoet/TypeSpecTest.kt

```diff
@@ -1051,6 +1051,50 @@ class TypeSpecTest {
         |""".trimMargin())
   }
 
+  @Test fun varargsNotLast() {
+    val taqueria = TypeSpec.classBuilder("Taqueria")
+        .addFun(FunSpec.builder("prepare")
+            .addParameter("workers", Int::class)
+            .addParameter("jobs", Runnable::class.asClassName(), VARARG)
+            .addParameter("start", Boolean::class.asClassName())
+            .build())
+        .build()
+    assertThat(toString(taqueria)).isEqualTo("""
+        |package com.squareup.tacos
+        |
+        |import java.lang.Runnable
+        |import kotlin.Boolean
+        |import kotlin.Int
+        |
+        |class Taqueria {
+        |  fun prepare(workers: Int, vararg jobs: Runnable, start: Boolean) {
+        |  }
+        |}
+        |""".trimMargin())
+  }
+
+  @Test fun multipleVarargs() {
+    val taqueria = TypeSpec.classBuilder("Taqueria")
+        .addFun(FunSpec.builder("prepare")
+            .addParameter("workers", Int::class)
+            .addParameter("jobs", Runnable::class.asClassName(), VARARG)
+            .addParameter("names", String::class.asClassName(), VARARG)
+            .build())
+        .build()
+    assertThat(toString(taqueria)).isEqualTo("""
+        |package com.squareup.tacos
+        |
+        |import java.lang.Runnable
+        |import kotlin.Int
+        |import kotlin.String
+        |
+        |class Taqueria {
+        |  fun prepare(workers: Int, vararg jobs: Runnable, vararg names: String) {
```

Comment:
The compiler will enforce it so i'm not sure it's worth duplicating. Especially since it'd have to be done on every object that collects `ParameterSpec`s